Dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/DartVmServiceDebugProcess.java

```diff
@@ -279,13 +286,13 @@ public void guessRemoteProjectRoot(@NotNull final ElementList<LibraryRef> librar
 
         LOG.assertTrue(file.getPath().startsWith(localProjectRoot.getPath() + "/"), file.getPath() + "," + localProjectRoot.getPath());
         final String relPath = file.getPath().substring(localProjectRoot.getPath().length()); // starts with slash
-        if (!relPath.startsWith("/lib/") && remoteUri.endsWith(relPath)) {
+        if (relPath.startsWith("/lib/") && remoteUri.endsWith(relPath)) {
           howManyFilesMatch++;
           myRemoteProjectRootUri = remoteUri.substring(0, remoteUri.length() - relPath.length());
         }
       }
 
-      if (howManyFilesMatch == 1) {
+      if (howManyFilesMatch > 0) {
```

Comment:
Shouldn't we be looking for `bar/main.dart` in that case (i.e., something relative to the local project root)? I worry that we'll reject a match and come up with no value for `myRemoteProjectRootUri`.